addons/website_forum/controllers/main.py

```diff
@@ -237,7 +237,10 @@ def question(self, forum, question, **post):
             'reversed': reversed,
         })
         if (request.httprequest.referrer or "").startswith(request.httprequest.url_root):
-            values['back_button_url'] = request.httprequest.referrer
+            if request.httprequest.referrer == request.httprequest.url_root + "web":
+                values['back_button_url'] = "/forum/%s" % slug(forum)
```

Comment:
that would be `forum._compute_website_url()` but @qsm-odoo is suggesting another idea which I agree with.
same fix as in master but keep the same value/key in dict (so it doesn't break anything without a `-u`).